lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -166,19 +166,19 @@ def verify(payment, options = {})
 
       def application_fee_from_response(response)
         return unless response.success?
-
-        application_fees = response.params["data"].select { |fee| fee["object"] == "application_fee" }
-        application_fees.first["id"] unless application_fees.empty?
+        response.params["application_fee"] unless response.params["application_fee"].empty?
       end
 
       def refund_application_fee(money, identification, options = {})
         return Response.new(false, "Application fee id could not be found") unless identification
 
         post = {}
         add_amount(post, money, options)
-        options.merge!(:key => @fee_refund_api_key)
+        options.merge!(:key => @fee_refund_api_key) if @fee_refund_api_key
+        options.delete(:stripe_account) if options[:stripe_account]
```

Comment:
Strictly speaking, `a_hash.delete(:no_such_key)` is safe, so you don't need the guard.